File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjJavaFX/Persistence/src/main/java/app/persistence/repository/jdbc/TeamsDBRepository.java
package app.persistence.repository.jdbc;

import app.model.Team;
import app.persistence.TeamsRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

@Component
public class TeamsDBRepository implements TeamsRepository {

    private final JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public TeamsDBRepository(Properties props){
        logger.info("Initializing TeamsDBRepository with properties: {} ", props);
        dbUtils = new JdbcUtils(props);
    }

    public TeamsDBRepository(){
        Properties props = new Properties();
        try {
            props.load(EmployeesDBRepository.class.getResourceAsStream("/bd.properties"));
            System.out.println("Server properties set. ");
            props.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find server.properties " + e);
            dbUtils = null;
            return;
        }
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public void add(Team elem) {
        logger.traceEntry("saving task {} ", elem);
        Connection con = dbUtils.getConnection();

        try(PreparedStatement preStm = con.prepareStatement("INSERT INTO Teams (name) VALUES (?);")){
            preStm.setString(1, elem.getName());
            int result = preStm.executeUpdate();
            logger.trace("Saved {} instances", result);
        }
        catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Long id) {
        logger.traceEntry("deleting task {} ", id);
        Connection con = dbUtils.getConnection();

        try(PreparedStatement preStm = con.prepareStatement("DELETE FROM Teams WHERE id = ?;")){
            preStm.setInt(1, Math.toIntExact(id));
            int result = preStm.executeUpdate();
            logger.trace("Deleted {} instances", result);
        }
        catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public void update(Long aLong, Team elem) {
        logger.traceEntry("updating task {} ", elem);
        Connection con = dbUtils.getConnection();

        try(PreparedStatement preStm = con.prepareStatement("UPDATE Teams SET name = ? WHERE id = ?;")){
            preStm.setString(1, elem.getName());
            preStm.setInt(2, Math.toIntExact(aLong));
            int result = preStm.executeUpdate();
            logger.trace("Updated {} instances", result);
        }
        catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
    }

    @Override
    public Team findById(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        Team team = null;

        try(PreparedStatement preStm = con.prepareStatement("SELECT * FROM Teams WHERE id = ? LIMIT 1;")){
            preStm.setInt(1, Math.toIntExact(aLong));
            try(ResultSet result = preStm.executeQuery()){
                if(result.next()){
                    Long id = (long) result.getInt("id");
                    String name = result.getString("name");
                    team = new Team(name);
                    team.setId(id);
                }
            }
        }
        catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return team;
    }

    @Override
    public Iterable<Team> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Team> teams = new ArrayList<>();

        try(PreparedStatement preStm = con.prepareStatement("SELECT * FROM Teams;")){
            try(ResultSet result = preStm.executeQuery()){
                while(result.next()){
                    Long id = (long) result.getInt("id");
                    String name = result.getString("name");
                    Team team = new Team(name);
                    team.setId(id);
                    teams.add(team);
                }
            }
        }
        catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        logger.traceExit();
        return teams;
    }
}

File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjCSharp/Persistence/db/TeamsDBRepository.cs
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SQLite;
using Model;

namespace Persistence.db
{
    public class TeamsDBRepository : TeamsRepository
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(EmployeesDBRepository));
        DBUtils dbUtils;

        public TeamsDBRepository()
        {
            logger.Info("Initializing TeamsDBRepository");
            dbUtils = new DBUtils();
        }

        public void add(Team elem)
        {
            logger.Info($"saving employee {elem}");
            IDbConnection con = dbUtils.GetConnection();

            try
            {
                using (IDbCommand cmd = con.CreateCommand())
                {
                    cmd.CommandText = "INSERT INTO Teams (name) VALUES (@fname);";
                    IDbDataParameter paramName = cmd.CreateParameter();
                    paramName.ParameterName = "@name";
                    paramName.Value = elem.Name;
                    cmd.Parameters.Add(paramName);

                    var result = cmd.ExecuteNonQuery();
                    if (result == 0)
                        throw new RepositoryException("No team added!");
                    logger.Info($"saved {result} instances");
                }
            }
            catch(SQLiteException ex)
            {
                logger.Error(ex.Message);
                Console.WriteLine("Error DB " + ex.Message);
            }
        }

        public void delete(long id)
        {
            logger.Info($"deleting team {id}");
            IDbConnection con = dbUtils.GetConnection();

            try
            {
                using (IDbCommand cmd = con.CreateCommand())
                {
                    cmd.CommandText = "DELETE FROM Teams WHERE id = @id;";
                    IDbDataParameter paramId = cmd.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    cmd.Parameters.Add(paramId);

                    var dataR = cmd.ExecuteNonQuery();
                    if (dataR == 0)
                        throw new RepositoryException("No team deleted!");
                    logger.Info("exit delete TeamsRepo");
                }
            }
            catch (SQLiteException ex)
            {
                logger.Error(ex.Message);
                Console.WriteLine("Error DB " + ex.Message);
            }
        }

        public IEnumerable<Team> findAll()
        {
            logger.Info("enter findAll");
            IDbConnection con = dbUtils.GetConnection();
            IList<Team> teams = new List<Team>();

            try
            {
                using (IDbCommand cmd = con.CreateCommand())
                {
                    cmd.CommandText = "SELECT id, name FROM Teams;";

                    using (var result = cmd.ExecuteReader())
                    {
                        while (result.Read())
                        {
                            long id = result.GetInt64(0);
                            String name = result.GetString(1);
                            Team t = new Team(name);
                            t.Id = id;
                            teams.Add(t);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            logger.Info("exit findAll");
            return teams;
        }

        public Team findById(long id)
        {
            logger.Info("enter findById");
            IDbConnection con = dbUtils.GetConnection();
            Team team = null;

            try
            {
                using (IDbCommand cmd = con.CreateCommand())
                {
                    cmd.CommandText = "SELECT id, name FROM Teams WHERE id = @id LIMIT 1;";
                    IDbDataParameter paramId = cmd.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    cmd.Parameters.Add(paramId);

                    using (var result = cmd.ExecuteReader())
                    {
                        if (result.Read())
                        {
                            long idT = result.GetInt64(0);
                            String name = result.GetString(1);
                            team = new Team(name);
                            team.Id = idT;
                        }
                    }
                    // log.InfoFormat("Exiting findOne with value {0}", null);
                    return team;
                }
            }
            catch(SQLiteException ex)
            {
                logger.Error(ex.Message);
                Console.WriteLine("Error DB " + ex.Message);
                return team;
            }
        }

        public void update(Team elem, long id)
        {
            logger.Info($"updating employee {elem}");
            IDbConnection con = dbUtils.GetConnection();

            try
            {
                using (IDbCommand cmd = con.CreateCommand())
                {
                    cmd.CommandText = "UPDATE Teams SET name = @name WHERE id = @id;";
                    IDbDataParameter paramName = cmd.CreateParameter();
                    paramName.ParameterName = "@name";
                    paramName.Value = elem.Name;
                    cmd.Parameters.Add(paramName);

                    IDbDataParameter paramId = cmd.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = elem.Id;
                    cmd.Parameters.Add(paramId);

                    var result = cmd.ExecuteNonQuery();
                    logger.Info($"updated {result} instances");
                }
            }
            catch (SQLiteException ex)
            {
                logger.Error(ex.Message);
                Console.WriteLine("Error DB " + ex.Message);
            }
        }
    }
}
